sdk/src/Core/Amazon.Runtime/Pipeline/RetryHandler/RetryPolicy.cs

```diff
@@ -446,26 +445,32 @@ private bool IsClockskew(IExecutionContext executionContext, Exception exception
                     // using accurate server time, calculate correction if local time is off
                     serverTime = serverTime.ToUniversalTime();
                     var diff = correctedNow - serverTime;
+                    var newCorrection = serverTime - realNow;
                     var absDiff = diff.Ticks < 0 ? -diff : diff;
-                    if (absDiff > clockSkewMaxThreshold)
-                    {
-                        var newCorrection = serverTime - realNow;
-                        Logger.InfoFormat(clockSkewMessageFormat,
-                            realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
 
-                        // Always set the correction, for informational purposes
-                        CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
+                    Logger.InfoFormat(clockSkewMessageFormat,
+                        realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
 
-                        var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+                    // Always set the correction, for informational purposes
+                    CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
 
-                        // Only retry if clock skew correction is not disabled
-                        if (shouldRetry)
-                        {
-                            // Set clock skew correction
-                            Logger.InfoFormat(clockSkewUpdatedFormat, newCorrection, endpoint);
-                            executionContext.RequestContext.IsSigned = false;
-                            return true;
-                        }
+                    // Some services such as S3 will return a generic 403 error for clock skew errors on HEAD requests
+                    // with no error code. In this case we will always retry the request if the clock skew is greater than
+                    // the threshold.
+                    if (isHead && absDiff > clockSkewMaxThreshold)
+                    {
+                        return true;
+                    }
+
+                    var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue && isClockskewErrorCode;
```

Comment:
I think this should be moved up before the prior if so it can be included there, removing the isClockskewErrorCode then change the if(shouldRetry) to an else

Reply:
I ended up merging the if statements to make it cleaner and moving the Shouldretry logic up.